src/gamecontroller.cs

```diff
@@ -535,4 +621,1162 @@ interface GCEventViewController {
 	}
+
+	[TV (14,0), Mac (11,0), iOS (14,0)]
+	[BaseType (typeof( NSObject))]
+	[DisableDefaultCtor]
+	interface GCColor : NSCopying, NSSecureCoding
+	{
+		[Export ("initWithRed:green:blue:")]
+		IntPtr Constructor (float red, float green, float blue);
+
+		[Export ("red")]
+		float Red { get; }
+
+		[Export ("green")]
+		float Green { get; }
+
+		[Export ("blue")]
+		float Blue { get; }
+	}
+
+	delegate void GCControllerTouchpadHandler (GCControllerTouchpad touchpad, float xValue, float yValue, float buttonValue, bool buttonPressed);
+
+	[TV (14,0), Mac (11,0), iOS (14,0)]
+	[BaseType (typeof (GCControllerElement))]
+	interface GCControllerTouchpad
+	{
+		[Export ("button")]
+		GCControllerButtonInput Button { get; }
+
+		[NullAllowed, Export ("touchDown", ArgumentSemantic.Copy)]
+		GCControllerTouchpadHandler TouchDown { get; set; }
+
+		[NullAllowed, Export ("touchMoved", ArgumentSemantic.Copy)]
+		GCControllerTouchpadHandler TouchMoved { get; set; }
+
+		[NullAllowed, Export ("touchUp", ArgumentSemantic.Copy)]
+		GCControllerTouchpadHandler TouchUp { get; set; }
+
+		[Export ("touchSurface")]
+		GCControllerDirectionPad TouchSurface { get; }
+
+		[Export ("touchState")]
+		GCTouchState TouchState { get; }
+
+		[Export ("reportsAbsoluteTouchSurfaceValues")]
+		bool ReportsAbsoluteTouchSurfaceValues { get; set; }
+
+		[Export ("setValueForXAxis:yAxis:touchDown:buttonValue:")]
+		void SetValue (float xAxis, float yAxis, bool touchDown, float buttonValue);
+	}
+
+	[TV (14,0), Mac (11,0), iOS (14,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface GCDeviceBattery
+	{
+		[Export ("batteryLevel")]
+		float BatteryLevel { get; }
+
+		[Export ("batteryState")]
+		GCDeviceBatteryState BatteryState { get; }
+	}
+
+	[TV (14,0), Mac (11,0), iOS (14,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface GCDeviceHaptics
+	{
+		[Export ("supportedLocalities", ArgumentSemantic.Strong)]
+		NSSet<NSString> SupportedLocalities { get; }
+
+		// Dependency on Xcode 12 CoreHaptics bindings
+		// -(CHHapticEngine * _Nullable)createEngineWithLocality:(GCHapticsLocality _Nonnull)locality;
+		//[Export ("createEngineWithLocality:")]
+		//[return: NullAllowed]
+		//CHHapticEngine CreateEngine (string locality);
+
+		[Field ("GCHapticsLocalityDefault")]
+		NSString LocalityDefault { get; }
+
+		[Field ("GCHapticsLocalityAll")]
+		NSString LocalityAll { get; }
+
+		[Field ("GCHapticsLocalityHandles")]
+		NSString LocalityHandles { get; }
+
+		[Field ("GCHapticsLocalityLeftHandle")]
+		NSString LocalityLeftHandle { get; }
+
+		[Field ("GCHapticsLocalityRightHandle")]
+		NSString LocalityRightHandle { get; }
+
+		[Field ("GCHapticsLocalityTriggers")]
+		NSString LocalityTriggers { get; }
+
+		[Field ("GCHapticsLocalityLeftTrigger")]
+		NSString LocalityLeftTrigger { get; }
+
+		[Field ("GCHapticsLocalityRightTrigger")]
+		NSString LocalityRightTrigger { get; }
+
+		[Field ("GCHapticDurationInfinite")]
+		float DurationInfinite { get; }
```

Comment:
I am not sure if we should remove the Haptic part, else I am unsure what infinite duration is it about

```suggestion
		[Field ("GCHapticDurationInfinite")]
		float HapticDurationInfinite { get; }
```

Reply:
+1 😄